### PR TITLE
GRANT-366: BCAT deployments

### PR DESCRIPTION
**Pull request by alva-fresh**
### Summary:

- Added a workflow file to deploy to dev;
- Added build config files for Openshft;
- Added `.pipeline` to configure [pipeline-cli](https://github.com/BCDevOps/pipeline-cli);

_Pending:_

- Workflows for `test` and `prod`;
- Confirm if we need `/configmaps/appsettings`;

### References:

- https://jira.th.gov.bc.ca/browse/GRANT-366

### Screenshots:

- _N/A_

### Safety

- [x] lint;
- [x] test;
- [x] prettier;

### CODE REVIEW

File: .pipeline/lib/config.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  dev: {
+    api_cpu: '100m',
+    api_memory: '512Mi',
+    changeId: updatedChangeId,
+    client_cpu: '100m',
+    client_memory: '512Mi',
+    host: `bcat-bfe2da-dev.apps.silver.devops.gov.bc.ca`,
+    instance: `${name}-dev-${updatedChangeId}`,
+    name: `${name}`,
+    namespace: 'bfe2da-dev',
```
Comment (karthikktamilmani-freshworks) [inline .pipeline/lib/config.js:36]: Make sure this is updated when we do UAT and PROD deployments!

File: .pipeline/lib/config.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    phase: 'dev',
```
Comment (karthikktamilmani-freshworks) [inline .pipeline/lib/config.js:37]: .

File: .pipeline/lib/config.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+const updatedChangeId = shortSha || changeId;
+
+Object.assign(options.git, { owner: 'bcgov', repository: 'BCAT' });
+const phases = {
+  build: {
+    changeId: updatedChangeId,
+    instance: `${name}-build-${updatedChangeId}`,
+    name: `${name}`,
+    namespace: 'bfe2da-tools',
```
Comment (karthikktamilmani-freshworks) [inline .pipeline/lib/config.js:20]: Could we assign `bfe2da` to a variable

**Review by karthikktamilmani-freshworks — APPROVED** [review]
Looks great!

I don't see config files for the deployment of the Database!

Also, `openshift/sysdig.yml` gives you access to the monitoring system!

**Comment by alva-fresh** [comment]
> Looks great!
> 
> I don't see config files for the deployment of the Database!
> 
> Also, `openshift/sysdig.yml` gives you access to the monitoring system!

Thanks @karthikktamilmani-freshworks! We don't have database deployment configs because it's managed by the gov team. We only need `api` and `client`.
It's a good idea to add `sysdig` but would need a confirmation to do so.

File: .pipeline/lib/config.js (hunk 3 above)
Comment (alva-fresh) [reply .pipeline/lib/config.js:20]: This might be easier to keep track of. I can update that, thanks for suggestion!

**Review by coryatwater-FW — APPROVED** [review]

**Pull request merged**